components/WeatherWidget/weatherData.ts

```diff
@@ -0,0 +1,84 @@
+import express from "express";
```

Comment:
Why did you create a  seperate server here?
Nextjs comes with a server. If you want to fetch data server side, use a api route.